File: 2021/1/part_1.nim
# nim doesn't like file that start with numbers, reasonable ig

import std/[strutils, sequtils]

let
    input = readFile("input.txt")
    lines: seq[string] = input.splitLines().toSeq

var
    increases = 0
    last = -1
    measurements = newSeq[int]()

for line in lines:
    try:
        measurements.add(line.parseInt)
    except ValueError:
        discard

last = measurements[0]

for m in measurements[1 .. measurements.high]:
    if m > last:
        increases += 1

    last = m

echo increases

File: 2021/1/part_1.js
const fs = require("fs");

const lines = fs.readFileSync("input.txt", "utf8").trim("\n").replace(/\r/g, "").split("\n");

const measurements = lines.map(n => parseInt(n, 10));

var last = measurements[0];
var increases = 0;

for (m of measurements.slice(1)) {
    if (m > last) increases++;

    last = m;
}

console.log(increases);
